tests/test_xmp.py

```diff
@@ -172,8 +172,8 @@ def test_dc_subject():
 def test_issue585():
     url = "https://github.com/mstamy2/PyPDF2/files/5536984/test.pdf"
     name = "mstamy2-5536984.pdf"
-    reader = PdfReader(BytesIO(get_pdf_from_url(url, name=name)))
     with pytest.raises(PdfReadError) as exc:
+        reader = PdfReader(BytesIO(get_pdf_from_url(url, name=name)))
```

Comment:
```suggestion
    reader = PdfReader(BytesIO(get_pdf_from_url(url, name=name)))
    with pytest.raises(PdfReadError) as exc:
```

Reply:
@pubpub-zz What was the reason to move this? I've just tested it locally; it seems not to be necessary to change the test

Reply:
I prefer to leave appart and clear any exception of PdfReader init and just focus on the exception raised by the extraction of metadata

Reply:
Let me check again; I thought there is no exception during init. If there is, then the second line is not necessary and should be removed (the `reader.xmp_metadata`)

Reply:
I've just ran it again. The exception happens in the `reader.xmp_metadata` line and thus only that line should be guarded by the `with pytest.raises ...` block. This way it is clear that it's that line which causes the exception.